presto-main/src/main/java/com/facebook/presto/type/TypeCoercer.java

```diff
@@ -349,6 +351,18 @@ private TypeCompatibility compatibility(Type fromType, Type toType)
         Type standardFromType = toStandardType(fromType);
         Type standardToType = toStandardType(toType);
 
+        if (standardFromType instanceof DistinctType && standardToType instanceof DistinctType) {
```

Comment:
Can we add comment to say we don't allow implicit cast for primitive types from/to distinct types?